File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const n = prices.length;
    const stack = [0];
    const answer = new Array(n).fill(0);
    
    for(let i = 0; i < n; i+=1) {
        while(stack.length && prices[stack[stack.length-1]] > prices[i]) {
            const top = stack.pop();
            answer[top] = i - top;
        }
        stack.push(i);
    }
    
    while(stack.length) {
        const top = stack.pop();
        answer[top] = n-1-top;
    }
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    prices = deque(prices)
    answer = []

    while prices:
        current_price = prices.popleft()
        price_not_fall_period = 0
        for next_price in prices:
            price_not_fall_period += 1
            if current_price > next_price:
                break

        answer.append(price_not_fall_period)

    return answer